test/test-app-folder-artifact-same.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+var assert = require('yeoman-assert');
+var constants = require('../generators/app/constants.js');
+var helpers = require('yeoman-test');
+var fs = require('fs');
+var os = require('os');
+var path = require('path');
+
+
+describe('generator-alfresco:app-folder-artifact-same', function () {
+
+  describe('artifactId != folder name for new project', function () {
+
+    this.timeout(60000);
+    this.osTempDir = path.join(os.tmpdir(), 'temp-test');
+
+    before(function (done) {
+      helpers.run(path.join(__dirname, '../generators/app'))
+        .inDir(this.osTempDir)
+        .withOptions({ 'skip-install': true })
+        .withPrompts({
+          sdkVersion: '2.1.1',
+          projectArtifactId: 'test-artifact',
+        })
+        .on('end', done);
+    }.bind(this));
+
+    it('creates files', function () {
+      assert.file([
+        '.yo-rc.json',
+      ]);
+    });
+    it('parent folder should equal to artifactId', function() {
+      assert.equal(path.basename(process.cwd()),'test-artifact');
+    });
+    it('grand-parent folder should be temp-test', function() {
+      assert.equal(path.basename(path.resolve('..')),'temp-test');
+    });
+    it('re-running the install with warning=true', function(done) {
```

Comment:
Rename? 'we can abort project creation based on artifactId and folder name mismatch'